aptos-move/e2e-tests/src/executor.rs

```diff
@@ -805,23 +951,71 @@ impl<O: OutputLogger> FakeExecutorImpl<O> {
         // TODO fetch values from state?
         let onchain_config = BlockExecutorConfigFromOnchain::on_but_large_for_test();
 
+        // Generate consecutive block metadata when using a shared cache (fuzzing/test).
+        // The same metadata is reused across sequential and parallel runs in comparison mode.
+        let metadata: Option<TransactionSliceMetadata> = self.get_txn_slice_metadata(mode);
+
+        let config = BlockExecutorConfig {
```

Comment:
can we make it `mut`, and set concurrency level below for parallel to `max(..)`? So that other field initialization is shared.